datamap/lucene/src/main/java/org/apache/carbondata/datamap/lucene/LuceneFineGrainDataMap.java

```diff
@@ -157,6 +162,14 @@ private String getQueryString(Expression expression) {
       return null;
     }
 
+    //maximum records to return
+    if (strQuery.contains(",") && strQuery.indexOf(",") != strQuery.length() - 1) {
+      max_doc = Integer.parseInt(strQuery.substring(strQuery.lastIndexOf(",") + 1).trim());
+      strQuery = strQuery.substring(0, strQuery.indexOf(","));
+    } else if (strQuery.contains(",") && strQuery.indexOf(",") == strQuery.length() - 1) {
+      strQuery = strQuery.substring(0, strQuery.indexOf(","));
```

Comment:
I am not sure why this if branch is needed

Reply:
If user doesnot specify the value in TEXT_MATCH UDF, like TEXT_MATCH("name:a", )  , default value should be Integer.MAX_VALUE